packages/client/src/LensClient.spec.ts

```diff
@@ -0,0 +1,99 @@
+import { Wallet } from 'ethers';
+
+import LensClient, { Credentials, mumbai, ReactionTypes } from '.';
+
+let credentials: Credentials;
+const getCredentials = async (wallet: Wallet) => {
+  if (credentials) {
+    return credentials;
+  }
+  const client = LensClient.init(mumbai);
+  const address = await wallet.getAddress();
+  const challenge = await client.generateChallenge(address);
+  const signature = await wallet.signMessage(challenge);
+
+  credentials = await client.authenticate(address, signature);
+  return credentials;
+};
+
+const getAuthenticatedClient = async (wallet: Wallet) => {
+  const client = LensClient.init(mumbai);
+  const address = await wallet.getAddress();
+  const challenge = await client.generateChallenge(address);
+  const signature = await wallet.signMessage(challenge);
+
+  await client.authenticate(address, signature);
+  return client;
+};
+
+describe(`Given a LensClient configured for testnet and a wallet`, () => {
+  const wallet = new Wallet(
+    'ca9a3a3d4026e6228713e683a9c45ef65a538b2f9336813bd597f5effa38668d', // found on reddit
+  );
+
+  describe(`when authenticating with wallet signature`, () => {
+    it(`authenticates with success`, async () => {
```

Comment:
nit: can we keep test description consistent with the rest of the codebase by using: `it('should ...`